pkg/webhook/clusterset/validatingWebhook.go

```diff
@@ -0,0 +1,356 @@
+	"net/http"
+)
+
+type AdmissionHandler struct {
+	KubeClient kubernetes.Interface
+}
+
+// toAdmissionResponse is a helper function to create an AdmissionResponse
+// with an embedded error
+func toAdmissionResponse(err error) *v1.AdmissionResponse {
+	return &v1.AdmissionResponse{
+		Result: &metav1.Status{
+			Message: err.Error(),
+		},
+	}
+}
+
+// admitFunc is the type we use for all of our validators and mutators
+type admitFunc func(request *v1.AdmissionRequest) *v1.AdmissionResponse
+
+// serve handles the http portion of a request prior to handing to an admit
+// function
+func (a *AdmissionHandler) serve(w io.Writer, r *http.Request, admit admitFunc) {
+	var body []byte
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+	// verify the content type is accurate
+	contentType := r.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		klog.Errorf("contentType=%s, expect application/json", contentType)
+		return
+	}
+
+	klog.V(2).Info(fmt.Sprintf("handling request: %s", body))
+
+	// The AdmissionReview that was sent to the webhook
+	requestedAdmissionReview := v1.AdmissionReview{}
+
+	// The AdmissionReview that will be returned
+	responseAdmissionReview := v1.AdmissionReview{}
+
+	deserializer := options.Codecs.UniversalDeserializer()
+	if _, _, err := deserializer.Decode(body, nil, &requestedAdmissionReview); err != nil {
+		klog.Error(err)
+		responseAdmissionReview.Response = toAdmissionResponse(err)
+	} else {
+		// pass to admitFunc
+		responseAdmissionReview.Response = admit(requestedAdmissionReview.Request)
+	}
+
+	responseAdmissionReview.Kind = requestedAdmissionReview.Kind
+	responseAdmissionReview.APIVersion = requestedAdmissionReview.APIVersion
+	// Return the same UID
+	responseAdmissionReview.Response.UID = requestedAdmissionReview.Request.UID
+
+	klog.V(2).Info(fmt.Sprintf("sending response: %+v", responseAdmissionReview))
+
+	respBytes, err := json.Marshal(responseAdmissionReview)
+	if err != nil {
+		klog.Error(err)
+	}
+	if _, err := w.Write(respBytes); err != nil {
+		klog.Error(err)
+	}
+}
+
+var managedClustersGVR = metav1.GroupVersionResource{
+	Group:    "cluster.open-cluster-management.io",
+	Version:  "v1",
+	Resource: "managedclusters",
+}
+var clusterDeploymentsGVR = metav1.GroupVersionResource{
+	Group:    "hive.openshift.io",
+	Version:  "v1",
+	Resource: "clusterdeployments",
+}
+var clusterPoolsGVR = metav1.GroupVersionResource{
+	Group:    "hive.openshift.io",
+	Version:  "v1",
+	Resource: "clusterpools",
+}
+var clusterClaimsGVR = metav1.GroupVersionResource{
+	Group:    "hive.openshift.io",
+	Version:  "v1",
+	Resource: "clusterclaims",
+}
+
+const (
+	clusterSetLabel = "cluster.open-cluster-management.io/clusterset"
+)
+
+func (a *AdmissionHandler) validateResource(request *v1.AdmissionRequest) *v1.AdmissionResponse {
+	status := &v1.AdmissionResponse{
+		Allowed: true,
+	}
+
+	switch request.Resource {
+	case managedClustersGVR:
```

Comment:
I think you can use 
```go
case managedClustersGVR, clusterDeploymentsGVR, clusterClaimsGVR....
```

Reply:
done. I also move the codes handling `clusterdeployment` created by `clusterclaim`  here.